storages/sled-storage/tests/sled_transaction.rs

```diff
@@ -408,144 +408,163 @@ async fn sled_transaction_gc() {
+        exec!(glue1 "CREATE TABLE TxGarlic (id2 INTEGER);");
+        exec!(glue1 "CREATE TABLE RealGarlic (id2 INTEGER);");
+
+        exec!(glue1 "BEGIN;");
+        exec!(glue1 "INSERT INTO TxGarlic VALUES (10);");
+        sleep();
+        exec!(glue2 "INSERT INTO TxGarlic VALUES (20);");
+        test!(glue1 "COMMIT;", Err(Error::StorageMsg("fetch failed - expired transaction has used (timeout)".to_owned())));
+        exec!(glue1 "ROLLBACK;");
+
+        // glue1 lock has expired, so TxGarlic table has only a single row (20)
+        test!(
+            glue1 "SELECT * FROM TxGarlic;",
+            Ok(select!(id2 I64; 20))
+        );
+
+        exec!(glue1 "BEGIN;");
+        exec!(glue1 "UPDATE TxGarlic SET id2 = id2 * 2;");
+        test!(
+            glue1 "SELECT * FROM TxGarlic;",
+            Ok(select!(id2 I64; 40))
+        );
+
+        // glue1 lock gets expired
+        sleep();
+
+        // glue1 tx must rollback
+        test!(
+            glue2 "SELECT * FROM TxGarlic;",
+            Ok(select!(id2 I64; 20))
+        );
+
+        exec!(glue1 "ROLLBACK;");
+        test!(
+            glue1 "SELECT * FROM TxGarlic;",
+            Ok(select!(id2 I64; 20))
+        );
+        test!(
+            glue2 "SELECT * FROM TxGarlic;",
+            Ok(select!(id2 I64; 20))
+        );
+
+        // UPDATE
+        exec!(glue1 "BEGIN;");
+        exec!(glue1 "UPDATE TxGarlic SET id2 = id2 + 1;");
+        sleep();
+        exec!(glue2 "BEGIN;");
+        exec!(glue2 "UPDATE TxGarlic SET id2 = id2 + 1;");
+        exec!(glue2 "ROLLBACK;");
+        exec!(glue1 "ROLLBACK;");
+
+        // DELETE
+        exec!(glue1 "BEGIN;");
+        exec!(glue1 "DELETE FROM TxGarlic;");
+        test!(glue1 "SELECT * FROM TxGarlic;", Ok(select!(id2)));
+        sleep();
+        test!(glue2 "SELECT * FROM TxGarlic;", Ok(select!(id2 I64; 20)));
+        exec!(glue2 "BEGIN;");
+        exec!(glue2 "DELETE FROM TxGarlic");
+        exec!(glue1 "ROLLBACK;");
+        exec!(glue2 "ROLLBACK;");
+        test!(glue1 "SELECT * FROM TxGarlic;", Ok(select!(id2 I64; 20)));
+        test!(glue2 "SELECT * FROM TxGarlic;", Ok(select!(id2 I64; 20)));
+
+        // DROP TABLE
+        exec!(glue2 "BEGIN;");
+        exec!(glue2 "DROP TABLE TxGarlic;");
+        sleep();
+        test!(glue2 "COMMIT;", Err(Error::StorageMsg("fetch failed - expired transaction has used (timeout)".to_owned())));
+        exec!(glue1 "DROP TABLE TxGarlic;");
+        test!(
+            glue1 "SELECT * FROM TxGarlic;",
+            Err(FetchError::TableNotFound("TxGarlic".to_owned()).into())
+        );
+        exec!(glue2 "ROLLBACK;");
+        test!(
+            glue2 "SELECT * FROM TxGarlic;",
+            Err(FetchError::TableNotFound("TxGarlic".to_owned()).into())
+        );
+    }
 }
 
 #[cfg(not(any(target_os = "macos", target_os = "ios")))]
 #[tokio::test]
 async fn sled_transaction_timeout_alter() {
+    use timeout_test::{sleep, TX_TIMEOUT};
```

Comment:
How about deleting it and importing it from the top of the `timeout_test` module..?
I think I'm writing everything.

Reply:
Thank you for the review, covered in 232f5a03c27a1a742b802c853bff5a4112c638e2